backend/internal/oauth/oauth2/granthandlers/tokenexchange.go

```diff
@@ -0,0 +1,470 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package granthandlers
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	appmodel "github.com/asgardeo/thunder/internal/application/model"
+	"github.com/asgardeo/thunder/internal/oauth/oauth2/constants"
+	"github.com/asgardeo/thunder/internal/oauth/oauth2/model"
+	"github.com/asgardeo/thunder/internal/system/config"
+	"github.com/asgardeo/thunder/internal/system/jwt"
+	"github.com/asgardeo/thunder/internal/system/log"
+	"github.com/asgardeo/thunder/internal/system/utils"
+)
+
+// tokenExchangeGrantHandler handles the token exchange grant type.
+type tokenExchangeGrantHandler struct {
+	jwtService jwt.JWTServiceInterface
+}
+
+// newTokenExchangeGrantHandler creates a new instance of tokenExchangeGrantHandler.
+func newTokenExchangeGrantHandler(
+	jwtService jwt.JWTServiceInterface,
+) GrantHandlerInterface {
+	return &tokenExchangeGrantHandler{
+		jwtService: jwtService,
+	}
+}
+
+// ValidateGrant validates the token exchange grant type request.
+func (h *tokenExchangeGrantHandler) ValidateGrant(tokenRequest *model.TokenRequest,
+	oauthApp *appmodel.OAuthAppConfigProcessedDTO) *model.ErrorResponse {
+	if constants.GrantType(tokenRequest.GrantType) != constants.GrantTypeTokenExchange {
+		return &model.ErrorResponse{
```

Comment:
IMO this should be invalid grant type or some similar error.
Is this the error returned in other grant handlers as well? In that case we should fix them all. 
If so, shall we track it as an improvement?